astro/src/components/bikes/SearchBikes.astro

```diff
@@ -1,15 +1,63 @@
 ---
 import { t, changeLanguage } from "i18next";
+import bikeModelListing from "@models/bike_model";
+import type { BikeModel } from "@app-types/bike_model";
+
+import SearchBikesVue from "@components/bikes/SearchBikes.vue";
+import { Debug } from "astro/components";
+import type { SearchOnImage } from "@app-types/search";
 
 export interface SearchBikesProps {
   lang: string;
+  type: SearchOnImage.SearchBikesType;
 }
 
-const { lang } = Astro.props as SearchBikesProps;
+const { lang, type = "listing" } = Astro.props as SearchBikesProps;
 changeLanguage(lang);
+
+const translations = {
+  brand: t("homepage.brand"),
+  model: t("homepage.model"),
+  serial: t("homepage.serial"),
+  description: t("homepage.description")
+};
+
+const bikeBrandModelListing: BikeModel.BikeBrandModelItem[] =
+  await bikeModelListing({
+    showBrand: true
+  });
+
+const reducedBrandsAndModels = bikeBrandModelListing
+  .map((item) => ({
+    model: item.name,
+    brand: item.bike_brand.name
+  }))
+  .reduce((acc: BikeModel.ReducedBrandsAndModels[], item) => {
+    const brand = item.brand;
+    const existingBrand = acc.find((entry) => entry?.brand === brand);
```

Comment:
correct me if i'm wrong, but it looks like that you are filtering a list locally, instead of sending filters to the server. If so, it's a wrong implementation